edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Hosting.cs

```diff
@@ -31,8 +35,47 @@ public class Hosting
             bool clientCertAuthEnabled,
             SslProtocols sslProtocols)
         {
+            /* Define the HttpsConnectionAdapterOptions which integrates with aspnetcore https middleware to help us manage the client cert chain.
+               We need to store the client cert chain in the connection context, so that we can perform custom CA validation later to authenticate with the service identity.
+               We are using logic here that depends on an undocumented api, however eliminating this dependency would require a large refactor.
+               Context in this issue: https://github.com/dotnet/aspnetcore/issues/21606
+               Link to aspnet https middleware: https://github.com/dotnet/aspnetcore/blob/e81033e094d4663ffd227bb4aed30b76b0631e6d/src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs */
+            HttpsConnectionAdapterOptions connectionAdapterOptions = new HttpsConnectionAdapterOptions()
+            {
+                /* ClientCertificateMode.AllowCertificate is the intuitive choice here, however is not what we need
+                   If we use this then aspnetcore will register it's own certificate validation callback when creating the SslStream.
+                   This aspnetcore-created certificate validation callback in turn calls the HttpsConnectionAdapterOptions.ClientCertificateValidation, which we can configure In HttpsConnectionAdapterOptions.
+                   This is not the desired solution for us because HttpsConnectionAdapterOptions.ClientCertificateValidation does not provide access to the client cert chain when the connection is in scope.
+
+                   Thus, we need to pass NoCertificate to block aspnet from creating the SslStream with the certificate validation callback.
+                   This will let us use HttpsConnectionAdapterOptions.OnAuthenticate in order to register certificate validation with the connection in scope.
+                   We can handle the cert / no-cert case inside of OnAuthenticate */
+                ClientCertificateMode = ClientCertificateMode.NoCertificate,
+                ServerCertificate = serverCertificate,
+                SslProtocols = sslProtocols,
+                OnAuthenticate = (context, options) =>
+                {
+                    if (clientCertAuthEnabled)
+                    {
+                        /* Certificate authorization can be enabled in edgeHub, however direct methods will hit this logic and not need certificates.
```

Comment:
The way I understand it is that direct method logic will hit this but still work without certificates because the RemoteCertificateValidationCallback is set and always returns true, so I think the comment should be where return true is and explaining that.